### ISSUE TITLE
Inflated iOS sessions in GA4/Firebase potentially caused by silent push launching the app in the background

### ISSUE THREAD

**Issue by jacao** [firebase/firebase-ios-sdk#12347]
### Description

We've noticed a significant surge in iOS user sessions within Google Analytics 4, primarily originating from Firebase (the Firebase SDK integrated into our iOS app). However, the recorded session counts seem disproportionate compared to other metrics we track. This inconsistency has led us to suspect that our session counts might be inflated.

One hypothesis I've been exploring relates to the behaviour of our app when it enters the background and iOS terminates it, Subsequently upon receiving a silent push notification iOS relaunches the app in the background, triggering the standard lifecycle functions of the AppDelegate, including application(_:didFinishLaunchingWithOptions:).

Despite thorough investigation, I've yet to find a resolution to this issue. While researching, I've encountered similar issues raised by others (now closed), although not specifically related to silent push notifications. One such discussion can be found here: https://github.com/firebase/firebase-ios-sdk/issues/6161. I'm curious if the iOS Firebase SDK has any logic to determine whether the app was launched via silent push and prevent any `session_start` events being fired.

Your insights or suggestions on addressing this matter would be greatly appreciated. Thank you.

### Reproducing the issue

Because of the nature of the issue I have found it really difficult to reproduce the issue. I have been able to reproduce the app being launched in the background via silent push by following these steps:

Using a physical device that already has the app you want to test installed on (and registered for push)

- Select Your Scheme: At the top left corner of the Xcode window, you'll see a toolbar with your project name and a "scheme" dropdown menu next to it. Click on the scheme dropdown menu and select "Edit Scheme..." from the dropdown list.
- Choose Run Configuration: In the Scheme editor window that appears, make sure "Run" is selected on the left sidebar.
- Edit Run Options: In the main area of the Scheme editor window, select the "Info" tab.
- Change Launch Behavior: Under the "Launch" section, you'll see a dropdown menu labeled "Launch". By default, it's set to "Automatically". Change this to "Wait for executable to be launched".
- Close Scheme Editor: Click the "Close" button to save your changes and close the Scheme editor window.
- Run Your App: Now when you run your app from Xcode (by clicking the play button or using the shortcut Command + R), Xcode will wait for the app to be launched before attaching the debugger.
- Send Silent Push: Send a silent push to the device that the app is being ran on. If this has successfully been received in Xcode you should see the app's status switch to running rather than waiting to launch/execute.

I have found it difficult to get the Firebase DebugView to work properly with this method (it is very inconsistent). Along with this I have to wait for the session to end before I can test again so its a timely process to try and debug the issue.

### Firebase SDK Version

10.2.0

### Xcode Version

15.1

### Installation Method

Swift Package Manager

### Firebase Product(s)

Analytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by Reedyuk** [comment]
Could this be the same issue as https://github.com/firebase/firebase-ios-sdk/issues/12219

**Comment by jacao** [comment]
> Could this be the same issue as #12219

Yes this looks like the exact same issue! I think that there is potentially some missing logic around the app being launched from a silent push causing the app to send a `session_start` event.

**Comment by jacao** [comment]
Does anyone have any thoughts on this?

**Comment by javierservatepoq** [comment]
We are experiencing the same. Any luck on it?

**Comment by nilesh-central** [comment]
We are experiencing the same, any luck guys?

**Comment by jacao** [comment]
No luck so far. We have raised with Firebase through various channels but haven't been able to get this resolved yet. Probably worth reaching out to Firebase/Google about this issue also.

**Comment by htcgh** [comment]
Hi all. My team is still evaluating various design options to resolve this issue. We appreciate your patience.

**Comment by nilesh-central** [comment]
@htcgh @tsunghung any updates bro on this? it's affecting lots of businesses all over the world I guess.

**Comment by htcgh** [comment]
We've largely aligned on a design option and are working on the implementation. We expect to have this out early next year.

**Comment by jwoo92** [comment]
@htcgh Thank you for your ongoing communication regarding this bug.

Is now a convenient time to check on the fix's progress? My team is also seeing a similar issue on iOS, resulting in inflated session counts.

I appreciate the team's and your work in maintaining the SDK and resolving this problem.

**Comment by htcgh** [comment]
Hi @jwoo92. Thanks for checking in. We're still in the process of testing this fix alongside another related change. Assuming that all tests pass, we hope to include the fix in the next upcoming SDK release. Thank you for your patience!

**Comment by htcgh** [comment]
Hi all. We released the fix in Firebase v11.9.0. Closing this ticket for now, but please reopen if the issue still persists after updating.
